File: generate-images.py
#!/usr/bin/env python3
"""
Larisha Nigerian Fashion - Professional Image Generator
Generates high-quality product and lifestyle images for the website
"""

import os
from PIL import Image, ImageDraw, ImageFilter
import random
import math

# Configuration
OUTPUT_DIR = "public/images"
IMAGE_WIDTH = 1200
IMAGE_HEIGHT = 1500

# Create output directory if it doesn't exist
os.makedirs(OUTPUT_DIR, exist_ok=True)

# Color palettes for different product categories
PALETTES = {
    "ankara": [
        (255, 69, 0),      # Red-Orange
        (184, 134, 11),    # Dark Goldenrod
        (25, 25, 112),     # Deep Blue
        (220, 20, 60),     # Crimson
    ],
    "lace": [
        (255, 192, 203),   # Pink
        (255, 215, 0),     # Gold
        (230, 230, 250),   # Lavender
    ],
    "bubu": [
        (128, 0, 32),      # Burgundy
        (211, 211, 211),   # Light Gray
        (169, 169, 169),   # Dark Gray
    ],
    "office": [
        (0, 51, 102),      # Dark Blue
        (192, 192, 192),   # Silver
        (64, 64, 64),      # Charcoal
    ],
    "traditional": [
        (184, 134, 11),    # Gold
        (139, 69, 19),     # Saddle Brown
        (210, 180, 140),   # Tan
    ],
    "neutral": [
        (255, 255, 255),   # White
        (128, 128, 128),   # Gray
        (0, 0, 0),         # Black
    ]
}

def create_gradient_background(width, height, color1, color2):
    """Create a gradient background"""
    image = Image.new('RGB', (width, height), color1)
    draw = ImageDraw.Draw(image, 'RGBA')
    
    # Create gradient
    r1, g1, b1 = color1
    r2, g2, b2 = color2
    
    for y in range(height):
        ratio = y / height
        r = int(r1 * (1 - ratio) + r2 * ratio)
        g = int(g1 * (1 - ratio) + g2 * ratio)
        b = int(b1 * (1 - ratio) + b2 * ratio)
        draw.rectangle([(0, y), (width, y+1)], fill=(r, g, b))
    
    return image

def add_pattern_overlay(image, pattern_type="geometric"):
    """Add decorative pattern overlay"""
    draw = ImageDraw.Draw(image, 'RGBA')
    width, height = image.size
    
    if pattern_type == "geometric":
        # Create geometric patterns
        step = 40
        alpha = 20
        color = (255, 255, 255, alpha)
        for x in range(0, width, step):
            draw.line([(x, 0), (x, height)], fill=color, width=1)
        for y in range(0, height, step):
            draw.line([(0, y), (width, y)], fill=color, width=1)
    
    elif pattern_type == "dots":
        # Create dot pattern
        radius = 8
        spacing = 50
        color = (255, 255, 255, 15)
        for x in range(0, width, spacing):
            for y in range(0, height, spacing):
                draw.ellipse([x-radius, y-radius, x+radius, y+radius], fill=color)
    
    elif pattern_type == "stripes":
        # Create stripe pattern
        stripe_width = 30
        color = (255, 255, 255, 10)
        for x in range(0, width, stripe_width * 2):
            draw.rectangle([(x, 0), (x + stripe_width, height)], fill=color)
    
    return image

def add_text_overlay(image, title, subtitle=""):
    """Add text overlay to image"""
    draw = ImageDraw.Draw(image)
    width, height = image.size
    
    # Add semi-transparent dark overlay for text readability
    overlay = Image.new('RGBA', (width, height), (0, 0, 0, 120))
    image.paste(overlay, (0, 0), overlay)
    
    # Add title
    title_bbox = draw.textbbox((0, 0), title)
    title_width = title_bbox[2] - title_bbox[0]
    title_x = (width - title_width) // 2
    title_y = height // 2 - 40
    
    draw.text((title_x, title_y), title, fill=(255, 255, 255), font=None)
    
    # Add subtitle if provided
    if subtitle:
        subtitle_bbox = draw.textbbox((0, 0), subtitle)
        subtitle_width = subtitle_bbox[2] - subtitle_bbox[0]
        subtitle_x = (width - subtitle_width) // 2
        subtitle_y = height // 2 + 20
        
        draw.text((subtitle_x, subtitle_y), subtitle, fill=(220, 220, 220), font=None)
    
    return image

def generate_product_image(product_name, category, is_hover=False):
    """Generate a single product image"""
    
    # Select color palette based on category
    if "ankara" in category.lower():
        palette = PALETTES["ankara"]
        pattern = "geometric"
    elif "lace" in category.lower():
        palette = PALETTES["lace"]
        pattern = "dots"
    elif "bubu" in category.lower():
        palette = PALETTES["bubu"]
        pattern = "stripes"
    elif "office" in category.lower():
        palette = PALETTES["office"]
        pattern = "geometric"
    elif "traditional" in category.lower():
        palette = PALETTES["traditional"]
        pattern = "geometric"
    else:
        palette = PALETTES["neutral"]
        pattern = "dots"
    
    # Create gradient background
    color1 = palette[0]
    color2 = palette[1] if len(palette) > 1 else palette[0]
    
    img = create_gradient_background(IMAGE_WIDTH, IMAGE_HEIGHT, color1, color2)
    
    # Add pattern overlay
    img = add_pattern_overlay(img, pattern)
    
    # Add text
    subtitle = "Hover View" if is_hover else "Product View"
    img = add_text_overlay(img, product_name[:30], subtitle)
    
    # Add subtle blur for premium feel
    img = img.filter(ImageFilter.GaussianBlur(radius=0.5))
    
    return img

def generate_collection_image(collection_name, is_story=False):
    """Generate collection or story section image"""
    
    # Define colors for each collection
    colors = {
        "ankara": ((255, 69, 0), (184, 134, 11)),
        "lace": ((255, 192, 203), (255, 215, 0)),
        "bubu": ((128, 0, 32), (211, 211, 211)),
        "office": ((0, 51, 102), (192, 192, 192)),
        "story": ((184, 134, 11), (220, 20, 60)),
    }
    
    key = "story" if is_story else collection_name.lower().split()[0]
    color1, color2 = colors.get(key, ((100, 100, 100), (200, 200, 200)))
    
    img = create_gradient_background(IMAGE_WIDTH, IMAGE_HEIGHT, color1, color2)
    img = add_pattern_overlay(img, "geometric")
    
    if is_story:
        img = add_text_overlay(img, "Nigerian Heritage", "Artisan Craftsmanship")
    else:
        img = add_text_overlay(img, collection_name, "Premium Collection")
    
    return img

def generate_hero_image():
    """Generate hero section image"""
    # Royal sapphire with gold
    img = create_gradient_background(IMAGE_WIDTH, IMAGE_HEIGHT, (25, 25, 112), (184, 134, 11))
    img = add_pattern_overlay(img, "geometric")
    img = add_text_overlay(img, "Ankara to Elegance", "Nigerian Fashion Pride")
    return img

def generate_queen_image(queen_number, description):
    """Generate queen/lifestyle image"""
    palettes = [
        ("ankara", ((255, 69, 0), (184, 134, 11))),
        ("lace", ((255, 192, 203), (255, 215, 0))),
        ("bubu", ((128, 0, 32), (211, 211, 211))),
        ("office", ((0, 51, 102), (192, 192, 192))),
        ("celebration", ((220, 20, 60), (184, 134, 11))),
    ]
    
    _, (color1, color2) = palettes[queen_number - 1]
    img = create_gradient_background(IMAGE_WIDTH, IMAGE_HEIGHT, color1, color2)
    img = add_pattern_overlay(img, "dots")
    img = add_text_overlay(img, f"Larisha Queen #{queen_number}", description)
    
    return img

def main():
    """Generate all images"""
    print("🎨 Generating Larisha Nigerian Fashion Images...")
    print(f"📁 Output directory: {OUTPUT_DIR}\n")
    
    # Track progress
    total = 0
    generated = 0
    
    try:
        # Hero image
        print("📸 Generating hero image...")
        hero = generate_hero_image()
        hero.save(os.path.join(OUTPUT_DIR, "hero-nigerian-fashion.jpg"), "JPEG", quality=95)
        print("   ✅ hero-nigerian-fashion.jpg")
        generated += 1
        total += 1
        
        # Story image
        print("\n📖 Generating story image...")
        story = generate_collection_image("story", is_story=True)
        story.save(os.path.join(OUTPUT_DIR, "story-nigerian-heritage.jpg"), "JPEG", quality=95)
        print("   ✅ story-nigerian-heritage.jpg")
        generated += 1
        total += 1
        
        # Collection images
        print("\n🎨 Generating collection images...")
        collections = ["ankara", "lace", "bubu", "office"]
        for col in collections:
            img = generate_collection_image(col)
            filename = f"collection-{col}.jpg"
            img.save(os.path.join(OUTPUT_DIR, filename), "JPEG", quality=95)
            print(f"   ✅ {filename}")
            generated += 1
            total += 1
        
        # Product images (25 products × 2 versions each = 50 images)
        print("\n👗 Generating product images (25 products × 2 versions)...")
        products = [
            ("Royal Sapphire Ankara Corset Dress", "ankara"),
            ("Sunset Gold Lace Gown", "lace"),
            ("Emerald Office Blazer", "office"),
            ("Burgundy Bubu Dream Dress", "bubu"),
            ("Crimson Lace & Ankara Blend Gown", "lace"),
            ("Sunshine Ankara Wrapper Set", "ankara"),
            ("Navy Office Wrap Trousers", "office"),
            ("Blush Pink Hand-Beaded Lace Dress", "lace"),
            ("Deep Forest Ankara Pencil Skirt", "ankara"),
            ("Cream Iro & Buba Traditional Set", "traditional"),
            ("Terracotta Ankara Blouse Peplum", "ankara"),
            ("Deep Purple Premium Lace Gown", "lace"),
            ("Multicolor Ankara Corset Top", "ankara"),
            ("Ivory Damask Agbada Dress", "traditional"),
            ("Gold Aso-Oke Pencil Skirt", "traditional"),
            ("Charcoal Ankara Smart Trousers", "ankara"),
            ("Navy Polished Linen Shift Dress", "neutral"),
            ("Ruby Red Ankara Ball Gown", "ankara"),
            ("Modern Prince Navy Agbada Set", "traditional"),
            ("Ankara Print Slim Fit Shirt Men", "ankara"),
            ("Emerald Ankara Shorts", "ankara"),
            ("Navy Guinea Brocade Sokoto", "traditional"),
            ("Rainbow Ankara Flare Dress Girls", "ankara"),
            ("Blush Lace Flower Girl Dress", "lace"),
            ("Ankara Cropped Blouse Skirt Set", "ankara"),
        ]
        
        for idx, (product_name, category) in enumerate(products, 1):
            # Main product view
            img_main = generate_product_image(product_name, category, is_hover=False)
            filename_main = f"product-{idx}a.jpg"
            img_main.save(os.path.join(OUTPUT_DIR, filename_main), "JPEG", quality=95)
            
            # Hover view (slightly different treatment)
            img_hover = generate_product_image(product_name, category, is_hover=True)
            filename_hover = f"product-{idx}b.jpg"
            img_hover.save(os.path.join(OUTPUT_DIR, filename_hover), "JPEG", quality=95)
            
            print(f"   ✅ product-{idx}a.jpg & product-{idx}b.jpg - {product_name}")
            generated += 2
            total += 2
        
        # Queen/Community images
        print("\n👑 Generating community images...")
        queens = [
            ("Ankara Corset Elegance", "In Style"),
            ("Lace Gown Magic", "Celebration"),
            ("Bubu Dreams", "Comfort Elegance"),
            ("Office Wear Power", "Professional"),
            ("Aso-Ebi Celebration", "Family Pride"),
        ]
        
        for idx, (title, desc) in enumerate(queens, 1):
            img = generate_queen_image(idx, desc)
            filename = f"queen-{idx}.jpg"
            img.save(os.path.join(OUTPUT_DIR, filename), "JPEG", quality=95)
            print(f"   ✅ {filename} - {title}")
            generated += 1
            total += 1
        
        # Summary
        print("\n" + "="*60)
        print(f"✨ IMAGE GENERATION COMPLETE!")
        print("="*60)
        print(f"📊 Generated: {generated}/{total} images")
        print(f"📁 Location: {os.path.abspath(OUTPUT_DIR)}")
        print("\n✅ All images are ready to display on your website!")
        print("\n💡 NEXT STEPS:")
        print("   1. Run: npm run dev")
        print("   2. Visit: http://localhost:3000")
        print("   3. Your website will display with these professional images")
        print("   4. Later, replace with real AI-generated images if desired")
        print("\n🎉 Your Larisha Nigerian Fashion site is now complete!")
        
    except Exception as e:
        print(f"\n❌ Error during image generation: {e}")
        import traceback
        traceback.print_exc()

if __name__ == "__main__":
    main()

File: generate-images.js
#!/usr/bin/env node

/**
 * Larisha Nigerian Fashion - SVG Image Generator
 * Generates professional-looking SVG images that render as JPG
 */

const fs = require("fs");
const path = require("path");

const OUTPUT_DIR = "public/images";

// Ensure output directory exists
if (!fs.existsSync(OUTPUT_DIR)) {
  fs.mkdirSync(OUTPUT_DIR, { recursive: true });
}

// Color palettes for different product categories
const PALETTES = {
  ankara: [
    "#FF4500", // Red-Orange
    "#B8860B", // Dark Goldenrod
    "#191970", // Deep Blue
    "#DC143C", // Crimson
  ],
  lace: [
    "#FFC0CB", // Pink
    "#FFD700", // Gold
    "#E6E6FA", // Lavender
  ],
  bubu: [
    "#800020", // Burgundy
    "#D3D3D3", // Light Gray
    "#A9A9A9", // Dark Gray
  ],
  office: [
    "#003366", // Dark Blue
    "#C0C0C0", // Silver
    "#404040", // Charcoal
  ],
  traditional: [
    "#B8860B", // Gold
    "#8B4513", // Saddle Brown
    "#D2B48C", // Tan
  ],
  neutral: [
    "#FFFFFF", // White
    "#808080", // Gray
    "#000000", // Black
  ],
};

/**
 * Create an SVG gradient
 */
function createGradient(color1, color2, id = "grad1") {
  return `
    <defs>
      <linearGradient id="${id}" x1="0%" y1="0%" x2="0%" y2="100%">
        <stop offset="0%" style="stop-color:${color1};stop-opacity:1" />
        <stop offset="100%" style="stop-color:${color2};stop-opacity:1" />
      </linearGradient>
    </defs>
  `;
}

/**
 * Create geometric pattern overlay
 */
function createGeometricPattern(width, height) {
  let pattern = "";
  const step = 40;

  // Vertical lines
  for (let x = 0; x < width; x += step) {
    pattern += `<line x1="${x}" y1="0" x2="${x}" y2="${height}" stroke="white" stroke-width="1" opacity="0.1"/>`;
  }

  // Horizontal lines
  for (let y = 0; y < height; y += step) {
    pattern += `<line x1="0" y1="${y}" x2="${width}" y2="${y}" stroke="white" stroke-width="1" opacity="0.1"/>`;
  }

  return pattern;
}

/**
 * Create dot pattern overlay
 */
function createDotPattern(width, height) {
  let pattern = "";
  const spacing = 50;

  for (let x = 0; x < width; x += spacing) {
    for (let y = 0; y < height; y += spacing) {
      pattern += `<circle cx="${x}" cy="${y}" r="8" fill="white" opacity="0.08"/>`;
    }
  }

  return pattern;
}

/**
 * Create stripe pattern overlay
 */
function createStripePattern(width, height) {
  let pattern = "";
  const stripeWidth = 30;

  for (let x = 0; x < width; x += stripeWidth * 2) {
    pattern += `<rect x="${x}" y="0" width="${stripeWidth}" height="${height}" fill="white" opacity="0.05"/>`;
  }

  return pattern;
}

/**
 * Generate a product image SVG
 */
function generateProductImageSVG(productName, category, isHover = false) {
  const width = 1200;
  const height = 1500;

  // Select palette
  let palette;
  let patternType = "geometric";

  if (category.toLowerCase().includes("ankara")) {
    palette = PALETTES.ankara;
    patternType = "geometric";
  } else if (category.toLowerCase().includes("lace")) {
    palette = PALETTES.lace;
    patternType = "dots";
  } else if (category.toLowerCase().includes("bubu")) {
    palette = PALETTES.bubu;
    patternType = "stripes";
  } else if (category.toLowerCase().includes("office")) {
    palette = PALETTES.office;
    patternType = "geometric";
  } else if (category.toLowerCase().includes("traditional")) {
    palette = PALETTES.traditional;
    patternType = "geometric";
  } else {
    palette = PALETTES.neutral;
    patternType = "dots";
  }

  const color1 = palette[0];
  const color2 = palette[1] || palette[0];

  let pattern = "";
  if (patternType === "geometric") {
    pattern = createGeometricPattern(width, height);
  } else if (patternType === "dots") {
    pattern = createDotPattern(width, height);
  } else if (patternType === "stripes") {
    pattern = createStripePattern(width, height);
  }

  const subtitle = isHover ? "Hover View" : "Product View";
  const truncatedName = productName.substring(0, 35);

  const svg = `<?xml version="1.0" encoding="UTF-8"?>
<svg width="${width}" height="${height}" xmlns="http://www.w3.org/2000/svg">
  ${createGradient(color1, color2)}
  
  <!-- Background -->
  <rect width="${width}" height="${height}" fill="url(#grad1)"/>
  
  <!-- Pattern overlay -->
  <g opacity="0.3">
    ${pattern}
  </g>
  
  <!-- Dark overlay for text -->
  <rect width="${width}" height="${height}" fill="black" opacity="0.35"/>
  
  <!-- Product name -->
  <text x="${width / 2}" y="${height / 2 - 40}" 
        font-family="serif" font-size="48" font-weight="bold" 
        text-anchor="middle" fill="white" letter-spacing="2">
    ${truncatedName}
  </text>
  
  <!-- Subtitle -->
  <text x="${width / 2}" y="${height / 2 + 40}" 
        font-family="sans-serif" font-size="24" 
        text-anchor="middle" fill="#E0E0E0">
    ${subtitle}
  </text>
  
  <!-- Decorative line -->
  <line x1="${width / 2 - 100}" y1="${height / 2 + 80}" 
        x2="${width / 2 + 100}" y2="${height / 2 + 80}" 
        stroke="white" stroke-width="2" opacity="0.5"/>
</svg>`;

  return svg;
}

/**
 * Generate a collection image SVG
 */
function generateCollectionImageSVG(collectionName, isStory = false) {
  const width = 1200;
  const height = 1500;

  const colors = {
    ankara: ["#FF4500", "#B8860B"],
    lace: ["#FFC0CB", "#FFD700"],
    bubu: ["#800020", "#D3D3D3"],
    office: ["#003366", "#C0C0C0"],
    story: ["#B8860B", "#DC143C"],
  };

  const key = isStory ? "story" : collectionName.toLowerCase().split()[0];
  const [color1, color2] = colors[key] || ["#646464", "#C8C8C8"];

  const title = isStory ? "Nigerian Heritage" : collectionName;
  const subtitle = isStory ? "Artisan Craftsmanship" : "Premium Collection";

  const svg = `<?xml version="1.0" encoding="UTF-8"?>
<svg width="${width}" height="${height}" xmlns="http://www.w3.org/2000/svg">
  ${createGradient(color1, color2)}
  
  <!-- Background -->
  <rect width="${width}" height="${height}" fill="url(#grad1)"/>
  
  <!-- Geometric pattern -->
  <g opacity="0.2">
    ${createGeometricPattern(width, height)}
  </g>
  
  <!-- Dark overlay -->
  <rect width="${width}" height="${height}" fill="black" opacity="0.3"/>
  
  <!-- Title -->
  <text x="${width / 2}" y="${height / 2 - 60}" 
        font-family="serif" font-size="64" font-weight="bold" 
        text-anchor="middle" fill="white" letter-spacing="3">
    ${title}
  </text>
  
  <!-- Subtitle -->
  <text x="${width / 2}" y="${height / 2 + 40}" 
        font-family="sans-serif" font-size="28" 
        text-anchor="middle" fill="#E0E0E0">
    ${subtitle}
  </text>
  
  <!-- Decorative elements -->
  <circle cx="${width / 2}" cy="${height - 150}" r="60" fill="white" opacity="0.1"/>
  <circle cx="100" cy="100" r="40" fill="white" opacity="0.1"/>
  <circle cx="${width - 100}" cy="${height - 100}" r="50" fill="white" opacity="0.1"/>
</svg>`;

  return svg;
}

/**
 * Generate hero image SVG
 */
function generateHeroImageSVG() {
  const width = 1200;
  const height = 1500;

  const svg = `<?xml version="1.0" encoding="UTF-8"?>
<svg width="${width}" height="${height}" xmlns="http://www.w3.org/2000/svg">
  ${createGradient("#191970", "#B8860B")}
  
  <!-- Background -->
  <rect width="${width}" height="${height}" fill="url(#grad1)"/>
  
  <!-- Pattern -->
  <g opacity="0.2">
    ${createGeometricPattern(width, height)}
  </g>
  
  <!-- Dark overlay -->
  <rect width="${width}" height="${height}" fill="black" opacity="0.25"/>
  
  <!-- Main title -->
  <text x="${width / 2}" y="${height / 2 - 100}" 
        font-family="serif" font-size="84" font-weight="bold" 
        text-anchor="middle" fill="white" letter-spacing="4">
    Ankara to Elegance
  </text>
  
  <!-- Subtitle -->
  <text x="${width / 2}" y="${height / 2 + 60}" 
        font-family="sans-serif" font-size="28" 
        text-anchor="middle" fill="#E0E0E0">
    Nigerian Fashion Pride
  </text>
  
  <!-- Decorative lines -->
  <line x1="${width / 4}" y1="${height / 2 + 150}" 
        x2="${width * 0.75}" y2="${height / 2 + 150}" 
        stroke="white" stroke-width="3" opacity="0.6"/>
</svg>`;

  return svg;
}

/**
 * Generate queen/community image SVG
 */
function generateQueenImageSVG(queenNumber, description) {
  const width = 1200;
  const height = 1500;

  const palettes = [
    ["#FF4500", "#B8860B"], // Ankara
    ["#FFC0CB", "#FFD700"], // Lace
    ["#800020", "#D3D3D3"], // Bubu
    ["#003366", "#C0C0C0"], // Office
    ["#DC143C", "#B8860B"], // Celebration
  ];

  const [color1, color2] = palettes[queenNumber - 1];

  const svg = `<?xml version="1.0" encoding="UTF-8"?>
<svg width="${width}" height="${height}" xmlns="http://www.w3.org/2000/svg">
  ${createGradient(color1, color2)}
  
  <!-- Background -->
  <rect width="${width}" height="${height}" fill="url(#grad1)"/>
  
  <!-- Dot pattern -->
  <g opacity="0.15">
    ${createDotPattern(width, height)}
  </g>
  
  <!-- Dark overlay -->
  <rect width="${width}" height="${height}" fill="black" opacity="0.3"/>
  
  <!-- Queen title -->
  <text x="${width / 2}" y="${height / 2 - 40}" 
        font-family="serif" font-size="52" font-weight="bold" 
        text-anchor="middle" fill="white" letter-spacing="2">
    Larisha Queen #${queenNumber}
  </text>
  
  <!-- Description -->
  <text x="${width / 2}" y="${height / 2 + 50}" 
        font-family="sans-serif" font-size="32" 
        text-anchor="middle" fill="#E0E0E0">
    ${description}
  </text>
  
  <!-- Decorative crown-like element -->
  <polygon points="${width / 2},${height / 2 - 120} ${width / 2 - 40},${height / 2 - 80} ${width / 2},${height / 2 - 60} ${width / 2 + 40},${height / 2 - 80}" 
           fill="white" opacity="0.3"/>
</svg>`;

  return svg;
}

/**
 * Save SVG as file
 */
function saveSVG(filename, svgContent) {
  const filepath = path.join(OUTPUT_DIR, filename);
  fs.writeFileSync(filepath, svgContent, "utf8");
  return filepath;
}

/**
 * Main function to generate all images
 */
async function main() {
  console.log("\n🎨 Generating Larisha Nigerian Fashion Images (SVG)...");
  console.log(`📁 Output directory: ${path.resolve(OUTPUT_DIR)}\n`);

  let generated = 0;

  try {
    // Hero image
    console.log("📸 Generating hero image...");
    const heroSVG = generateHeroImageSVG();
    saveSVG("hero-nigerian-fashion.svg", heroSVG);
    console.log("   ✅ hero-nigerian-fashion.svg");
    generated++;

    // Story image
    console.log("\n📖 Generating story image...");
    const storySVG = generateCollectionImageSVG("story", true);
    saveSVG("story-nigerian-heritage.svg", storySVG);
    console.log("   ✅ story-nigerian-heritage.svg");
    generated++;

    // Collections
    console.log("\n🎨 Generating collection images...");
    const collections = ["ankara", "lace", "bubu", "office"];
    for (const col of collections) {
      const colSVG = generateCollectionImageSVG(col);
      saveSVG(`collection-${col}.svg`, colSVG);
      console.log(`   ✅ collection-${col}.svg`);
      generated++;
    }

    // Products
    console.log("\n👗 Generating product images (25 products × 2 versions)...");
    const products = [
      ("Royal Sapphire Ankara Corset Dress", "ankara"),
      ("Sunset Gold Lace Gown", "lace"),
      ("Emerald Office Blazer", "office"),
      ("Burgundy Bubu Dream Dress", "bubu"),
      ("Crimson Lace & Ankara Blend Gown", "lace"),
      ("Sunshine Ankara Wrapper Set", "ankara"),
      ("Navy Office Wrap Trousers", "office"),
      ("Blush Pink Hand-Beaded Lace Dress", "lace"),
      ("Deep Forest Ankara Pencil Skirt", "ankara"),
      ("Cream Iro & Buba Traditional Set", "traditional"),
      ("Terracotta Ankara Blouse Peplum", "ankara"),
      ("Deep Purple Premium Lace Gown", "lace"),
      ("Multicolor Ankara Corset Top", "ankara"),
      ("Ivory Damask Agbada Dress", "traditional"),
      ("Gold Aso-Oke Pencil Skirt", "traditional"),
      ("Charcoal Ankara Smart Trousers", "ankara"),
      ("Navy Polished Linen Shift Dress", "neutral"),
      ("Ruby Red Ankara Ball Gown", "ankara"),
      ("Modern Prince Navy Agbada Set", "traditional"),
      ("Ankara Print Slim Fit Shirt Men", "ankara"),
      ("Emerald Ankara Shorts", "ankara"),
      ("Navy Guinea Brocade Sokoto", "traditional"),
      ("Rainbow Ankara Flare Dress Girls", "ankara"),
      ("Blush Lace Flower Girl Dress", "lace"),
      ("Ankara Cropped Blouse Skirt Set", "ankara"),
    ];

    for (let idx = 0; idx < products.length; idx++) {
      const [productName, category] = products[idx];

      // Main product view
      const mainSVG = generateProductImageSVG(productName, category, false);
      saveSVG(`product-${idx + 1}a.svg`, mainSVG);

      // Hover view
      const hoverSVG = generateProductImageSVG(productName, category, true);
      saveSVG(`product-${idx + 1}b.svg`, hoverSVG);

      console.log(
        `   ✅ product-${idx + 1}a.svg & product-${idx + 1}b.svg - ${productName.substring(0, 30)}`,
      );
      generated += 2;
    }

    // Queens
    console.log("\n👑 Generating community images...");
    const queens = [
      ("Ankara Corset Elegance", "In Style"),
      ("Lace Gown Magic", "Celebration"),
      ("Bubu Dreams", "Comfort Elegance"),
      ("Office Wear Power", "Professional"),
      ("Aso-Ebi Celebration", "Family Pride"),
    ];

    for (let idx = 0; idx < queens.length; idx++) {
      const [title, desc] = queens[idx];
      const queenSVG = generateQueenImageSVG(idx + 1, desc);
      saveSVG(`queen-${idx + 1}.svg`, queenSVG);
      console.log(`   ✅ queen-${idx + 1}.svg - ${title}`);
      generated++;
    }

    // Summary
    console.log("\n" + "=".repeat(70));
    console.log("✨ IMAGE GENERATION COMPLETE!");
    console.log("=".repeat(70));
    console.log(`📊 Generated: ${generated} SVG image files`);
    console.log(`📁 Location: ${path.resolve(OUTPUT_DIR)}`);
    console.log("\n✅ All SVG images are ready to display on your website!");
    console.log("\n💡 NEXT STEPS:");
    console.log(
      "   1. Update image references in components to use .svg instead of .jpg",
    );
    console.log("   2. Run: npm run dev");
    console.log("   3. Visit: http://localhost:3000");
    console.log(
      "   4. Your website will display with professional SVG images!",
    );
    console.log(
      "\n📝 SVG images are vector-based and will look crisp on all screen sizes",
    );
    console.log(
      "   You can later replace these with AI-generated JPEG images if needed.",
    );
    console.log("\n🎉 Your Larisha Nigerian Fashion site is ready to launch!");
  } catch (error) {
    console.error("\n❌ Error during image generation:", error);
    process.exit(1);
  }
}

main();
